### DISCUSSION TITLE
Using slint without cmake

### DISCUSSION THREAD

**Discussion by Frityet** [slint-ui/slint#5682] [General]
How could I use slint without cmake? I want to use it with [xmake](https://xmake.io). I don't see any offical support so im prepared to write my own rules to get this to work. Is there docs on the slint compiler?

**Comment by ogoffart** [comment]
There is no docs to use it with anything else than cmake. This is missing.
But this should be fairly easy to use from any build system:

 1. Install Slint from the binary packages, or from the source (using cmake / make install)
 2. The Slint package contains the `slint-compiler` executable in the bin folder. You shall call it from your build system to generate a header file. use `slint-compiler --help` for details. You probably want to invoke `$SLINT_PATH/bin/slint-compiler your-ui.slint -f cpp -o your-ui.h` or something like that
 3. Then add as include path the `$SLINT_PATH/include/slint` as well as the path where your-ui.h file was generated.
 4. Link against `$SLINT_PATH/lib/libslint_cpp.so` (or .dll)

That should be it.
